### ISSUE TITLE
ethon: NoMethodError: undefined method `enum' for Ethon::Curl:Module

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1524]
It seems, in this case it looks on a problem with metaprogramming and extending modules.

To reproduce

```
require 'ethon'
```

Error:

```
NoMethodError: undefined method `enum' for Ethon::Curl:Module
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon/curls/constants.rb:16:in `Curl'
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon/curls/constants.rb:2:in `Ethon'
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon/curls/constants.rb:1:in `<top (required)>'
	from ~/.rubies/truffleruby-1.0.0-rc10/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `gem_original_require'
	from ~/.rubies/truffleruby-1.0.0-rc10/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon/curl.rb:26:in `Curl'
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon/curl.rb:13:in `Ethon'
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon/curl.rb:8:in `<top (required)>'
	from ~/.rubies/truffleruby-1.0.0-rc10/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `gem_original_require'
	from ~/.rubies/truffleruby-1.0.0-rc10/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.gem/truffleruby/2.4.4/gems/ethon-0.11.0/lib/ethon.rb:15:in `<top (required)>'
	from ~/.rubies/truffleruby-1.0.0-rc10/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:130:in `gem_original_require'
	from ~/.rubies/truffleruby-1.0.0-rc10/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:130:in `require'
	from (irb):3
	from ~/.rubies/truffleruby-1.0.0-rc10/bin/irb:29:in `<main>'
```

**Comment by Vad1mo** [comment]
I am facing a similar or same problem 

```
Possibly unsupported Ruby implementation
/Users/v/.rvm/gems/truffleruby-1.0.0-rc10/gems/ethon-0.9.0/lib/ethon/curls/constants.rb:16:in `Curl': undefined method `enum' for Ethon::Curl:Module (NoMethodError)
```

**Comment by deepj** [comment]
@Vad1mo I doubt there is one. It looks an issue in TruffleRuby interpreter. Maybe @eregon could know more and what he thinks about this.

**Comment by eregon** [comment]
I'm looking at improving our FFI implementation, which is currently fairly minimal and causing this, see #1529. I'll keep this issue open to make sure `require "ethon"` works.

**Comment by eregon** [comment]
I fixed this in 59fb511e431b4795580ce7d191b84f0199351012 and added complete support for the FFI, with a pure-Ruby implementation using Truffle NFI.
I checked and `require "ethon"` now works.
It will be in the next release (1.0.0-rc16).
Thank you for the report.
